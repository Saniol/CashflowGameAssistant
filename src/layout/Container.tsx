import React, {useState, useCallback} from 'react';
import {View, SafeAreaView, StatusBar, StyleSheet} from 'react-native';
import {Header} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import {useHistory} from 'react-router-native';
import Lang from '../lang/pl';
import Backgrounds from '../global/Backgrounds';
import Colors from '../global/Colors';
import LayoutStyles from '../global/LayoutStyles';
import Menu from './Menu';

const ContainerStyles = StyleSheet.create({
    container: {
        flex: 1,
        height: '100%',
        position: 'relative',
    },
    header: {
        height: 50,
        paddingTop: 0,
    },
    title: {
        color: Colors.background,
        fontSize: 18,
        fontWeight: 'bold',
    },
});

const AppContainer: React.FC = ({children}) => {
    const [menuVisible, setMenuVisible] = useState(false);
    const history = useHistory();
    const goTo = useCallback(
        path => {
            history.push(path);
            setMenuVisible(false);
        },
        [history, setMenuVisible],
    );
    const currentPathname = history.location.pathname;

    return (
        <View style={[ContainerStyles.container, Backgrounds.light]}>
            <Header
                containerStyle={[ContainerStyles.header, Backgrounds.dark]}
                centerComponent={{
                    text: Lang.menuHeader,
                    style: ContainerStyles.title,
                }}
                rightComponent={
                    <Icon
                        testID="headerNavicon"
                        name={menuVisible ? 'times' : 'navicon'}
                        size={36}
                        color={Colors.light}
                        onPress={() => setMenuVisible(!menuVisible)}
                    />
                }
                leftComponent={
                    menuVisible || currentPathname === '/home' ? (
                        undefined
                    ) : (
                        <Icon
                            name="arrow-left"
                            size={24}
                            color={Colors.light}
                            onPress={() => goTo('/')}
                        />
                    )
                }
            />
            <StatusBar hidden />
            <SafeAreaView style={LayoutStyles.flexItem}>
                {menuVisible ? (
                    <Menu testID="mainMenu" goTo={goTo} />
                ) : (
                    children
                )}
            </SafeAreaView>
        </View>
    );
};

export default AppContainer;
