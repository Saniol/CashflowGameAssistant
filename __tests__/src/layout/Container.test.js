import {Text} from 'react-native';
import React from 'react';
import {NativeRouter} from 'react-router-native';
import {mount} from 'enzyme';
import Container from '../../../src/layout/Container';
import renderWithContexts from '../../common/renderWithContexts';

describe('<Container>', () => {
    renderWithContexts(Container);

    describe('rendered content', () => {
        let wrapper;

        beforeAll(() => {
            wrapper = mount(
                <NativeRouter>
                    <Container>
                        <Text testID="testChild" />
                    </Container>
                </NativeRouter>,
            );
        });

        const pressNavicon = () => {
            const navicon = wrapper
                .findWhere(node => node.prop('testID') === 'headerNavicon')
                .first();

            navicon.props().onPress();

            wrapper.update();
        };

        it('should display given children', () => {
            expect(
                wrapper.findWhere(node => node.prop('testID') === 'testChild'),
            ).toExist();
        });

        describe('navicon clicked', () => {
            it('should display <Menu>', () => {
                pressNavicon();
                expect(
                    wrapper.findWhere(
                        node => node.prop('testID') === 'mainMenu',
                    ),
                ).toExist();
            });
        });
        describe('navicon clicked again', () => {
            it('should display given children', () => {
                pressNavicon();
                expect(
                    wrapper.findWhere(
                        node => node.prop('testID') === 'testChild',
                    ),
                ).toExist();
            });
        });
    });
});
