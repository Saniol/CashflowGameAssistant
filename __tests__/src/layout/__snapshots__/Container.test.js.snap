// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": "100%",
        "position": "relative",
      },
      Object {
        "backgroundColor": "#F8E6B0",
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEB35F",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 0,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 50,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#F8E6B0",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#FCE9C4",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
        text="Cassflow Game Assistant"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Cassflow Game Assistant
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#F8E6B0",
              "fontSize": 36,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="headerNavicon"
      >
        
      </Text>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
